relay.go

```diff
@@ -654,6 +664,80 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	fragWriter := newFragmentingWriter(
+		NullLogger, r.newFragmentSender(relayToDest.destination, f, origID),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	arg2Writer, err := fragWriter.ArgWriter(f.isArg2Fragmented)
+	if err != nil {
+		return fmt.Errorf("get arg2 writer: %v", err)
+	}
+
+	if err := writeArg2WithAppends(arg2Writer, f.arg2(), f.arg2appends); err != nil {
+		return fmt.Errorf("write arg2: %v", err)
+	}
+	if f.isArg2Fragmented {
+		return nil
+	}
+
+	if err := NewArgWriter(fragWriter.ArgWriter(true)).Write(f.arg3()); err != nil {
+		return errors.New("arg3 write failed")
+	}
+
+	return nil
+}
+
+func writeArg2WithAppends(w io.WriteCloser, arg2 []byte, appends []keyVal) (err error) {
+	defer w.Close()
+
+	var nh uint16
+	if len(arg2) > 0 {
+		nh = binary.BigEndian.Uint16(arg2[:2])
+	}
+	nh += uint16(len(appends))
+	if err := writeUint16(w, nh); err != nil {
+		return fmt.Errorf("write arg2 nh: %v", err)
+	}
+
+	if len(arg2) > 0 {
+		if _, err := w.Write(arg2[2:]); err != nil {
```

Comment:
add comments to explain this code -- it makes sense to someone who has the protocol as top of mind, but will be hard to understand in a few months.